### PR TITLE
[search] Pre-localities selection is moved to LocalityScorer.

### PR DESCRIPTION
**Pull request by ygorshenin**
@bykoianko @mpimenov PTAL

### CODE REVIEW

**Comment by ygorshenin** [comment]
As benefits from https://github.com/mapsme/omim/pull/5292 are quite arguable, I decided to move out refactoring part from the pull request. Note that it's possible to simplify locality scorer tests further, all what we need to do that is to modify geocoder params to support number of localities geocoder may read, and start to generate mwms instead of self-created search index.

**Comment by ygorshenin** [comment]
But let's do all this testing-related stuff in the individual pull request.

File: search/geocoder.cpp [hunk 1]
```diff
@@ -745,7 +706,7 @@ void Geocoder::FillLocalitiesTable(BaseContext const & ctx)
 
 #if defined(DEBUG)
         ft.GetName(StringUtf8Multilang::kDefaultCode, city.m_defaultName);
-        LOG(LDEBUG, ("City =", city.m_defaultName, radius));
+        LOG(LINFO, ("City =", city.m_defaultName, "radius =", radius, "prob = ", city.m_prob));
```
Comment (mpimenov) [inline search/geocoder.cpp]: Лишний пробел в "prob = ".

Comment (ygorshenin) [inline search/geocoder.cpp]: Done

**Comment by bykoianko** [comment]
LGTM

**Comment by mpimenov** [comment]
LGTM

**Pull request merged**